### PR TITLE
Jenkins 40027 regression cant open branches that have / in the name

### PR DESCRIPTION
**Pull request by scherler**
# Description

See [JENKINS-40027](https://issues.jenkins-ci.org/browse/JENKINS-40027).

Rewrite of https://github.com/jenkinsci/blueocean-acceptance-test/pull/72

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@jenkinsci/code-reviewers @reviewbybees

### CODE REVIEW

**Comment by michaelneale** [comment]
@scherler what are all these other changes to this branch? It seems a bad merge?

**Comment by michaelneale** [comment]
@scherler I can't say I am a fan of this style of tests. It doesn't read like a script, but more like a program. This would be fine, except it makes it hard for others to follow the test, or, more importantly, to copy this approach to other tests. When I have tried to understand your test code before, I spent hours and got lost. 

So this is -1 on me for this style, I don't think this aids clarity. My version of it, which was inferior, had more repetition (that could have been put in a function) is something anyone can follow. I am not even sure this test is right, I have to take your word for it.

File: src/test/js/multibranch/multibranchOpening.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        git.createBranch('feature/1', pathToRepo)
+          .then(done);
+      });
+  },
+  /**
+   * Make sure we can open the master and the feature/1 branch results screen from activity and branch tab
+   * Regression: https://issues.jenkins-ci.org/browse/JENKINS-40027
+   */
+  'open master and feature/1 branch from activity and from branch tab': function (browser) {
+    const cases = [{
```
Comment (michaelneale) [inline src/test/js/multibranch/multibranchOpening.js]: this adds very little to the test. I am only using the partial id matching as a way to find a branch name in the list, but this elevates it to a higher level without actually explaining what the patterns are, this is less clear than before.

File: src/test/js/multibranch/multibranchOpening.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        regexp: 'tr[id^="master"]',
+        name: 'masterBranchesMB',
+        branch: true,
+      }, {
+        regexp: 'tr[id^="feature"]',
+        name: 'featureBranchesMB',
+        branch: true,
+      },
+    ];
+    async.mapSeries(cases, function (usecase, callback) {
```
Comment (michaelneale) [inline src/test/js/multibranch/multibranchOpening.js]: this is where most of the test happens, but now is an obscure named map. This is not friendly for others trying to follow the pattern. Its clever, and great production code, but it isn't an acceptance test

File: src/main/js/page_objects/blueocean/bluePipelineRunDetail.js [hunk 3]
```diff
@@ -110,6 +110,19 @@ module.exports.commands = [{
         // E.g. nothing on the pipeline graph that allows us to find it.
         return this;
     },
+    assertMultiBranchResult: function (callback) {
```
Comment (michaelneale) [inline src/main/js/page_objects/blueocean/bluePipelineRunDetail.js]: don't think this is sufficiently re-usable to be parked in here. It means you have to jump all over the code to get the slightest hint of what the test actually does .

**Comment by michaelneale** [comment]
overall - I like it (does it run when running on CI? I wasn't sure if there could be timing issues, as it looks for the spinner), but I am not a fan of this style of test code (would like others to talk about it, maybe I am in minority). I have struggled in the past to read the ATH code when I just wanted to see the basic assertions and css patterns it was looking for in response to actions.

File: src/main/js/page_objects/blueocean/bluePipelineRunDetail.js (hunk 3 above)
Comment (scherler) [inline src/main/js/page_objects/blueocean/bluePipelineRunDetail.js]: Actually that part it's a must. That belongs in the detail page. @tfennelly will surely agree

**Comment by scherler** [comment]
It it's your code but less lines. There is imo nothing bad in using a function to create the test cases ( which had been the same) I agree that i should have provided more comments in the code. Anyway for me that is the kind of code in ath i consider it very clean.

**Comment by michaelneale** [comment]
@tfennelly and I discussed this, going to move this stuff to the other PR to refactor out the common assertions

**Pull request closed without merging**